asciidoc/courses/gds-getting-started-with-graph-data-science/modules/2-gds-basic-concepts/lessons/1-understand-gds-workflow/lesson.adoc

```diff
@@ -0,0 +1,238 @@
+= Understand the GDS workflow
+:type: lesson
+:order: 1
+
+[.transcript]
+== The GDS workflow
+
+Every GDS analysis follows the same basic two or three-step pattern:
+
+**1. Project** - Create an in-memory graph  
+
+**2. Run** - Execute algorithms on the projection  
+
+**3. Write** - Store results back to the database
+
+This workflow separates your source data from analysis, enabling fast, iterative experimentation without modifying your original graph.
+
+The reason we specify 'two __or__ three' is that you can choose to run an algorithm and write its results in a single operation instead of two. Or, you can chain multiple algorithm results to feed a final algorithm before writing back to the graph.
+
+The reasons for this distinction will become clearer as you progress through the course. 
+
+
+== Why this workflow?
+
+**Speed** - In-memory projections are orders of magnitude faster than querying the database directly
+
+**Safety** - Your source graph remains unchanged until you explicitly write results back
+
+**Flexibility** - Run multiple algorithms on the same projection, compare results, iterate quickly
+
+
+== The three steps
+
+image::images/gdsworkflow.png[The gds workflow diagram going from main graph through projection to running an algorithm to writing back to the main graph (optional)]
+
+=== 1. Project a Graph
+The graph projected by GDS is an in-memory structure containing nodes and relationships -- just like your main graph. However, the in-memory graph you create is optimized for topology and property lookup operations.
+
+When you 'project' a graph, you are literally instantiating a new sub-graph in memory. 
+
+[source,cypher]
+----
+MATCH (source:Actor)-[r:ACTED_IN]->(target:Movie)     //(1)
+WITH gds.graph.project(                               //(2)
+  'actors-graph',
+  source,
+  target
+) AS g
+RETURN g.graphName AS graph, 
+  g.nodeCount AS nodes, 
+  g.relationshipCount AS rels                         //(3)
+----
+
+The graph projection above has three components:
+
+**1. Cypher query**
+
+This works in the same way as any other Cypher query you are used to running. However, instead of returning the graph results in the Neo4j Browser, you send it to GDS as a projection with the line that follows.
+
+**2. Call GDS**
+
+The GDS WITH statement works the same way as any WITH statement you are used to running in the Neo4j Browser. You call `gds.graph.project` to create the in-memory graph matched by the preceding Cypher statement.
```

Comment:
Prefer numbered lists and use a + at the end of the line for a line break

```
* **1. Cypher query** +
This works in the same way as any other Cypher query you are used to running. However, instead of returning the graph results in the Neo4j Browser, you send it to GDS as a projection with the line that follows.

* **2. Call GDS** +
The GDS WITH statement works the same way as any...
```